src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -70,6 +70,7 @@ private static bool IsValidRootConfigType(ITypeSymbol? type)
             {
                 if (type is null ||
                     type.SpecialType is SpecialType.System_Object or SpecialType.System_Void ||
+                    !IsAccessibleFromGenBinders(type) ||
```

Comment:
>!IsAccessibleFromGenBinders(type) || [](http://example.com/codeflow?start=20&length=36)

Is this mean we'll have a difference between the source gen and the runtime in such cases?

Reply:
In a sense... we won't generate a replacement interceptor for the invocation, meaning that it will always fallback to reflection. AOT users will get a warning and the fix would be to adjust the visibility.

Reply:
Will the users get anything to help them tell them what to do? I mean how the users know the source gen didn't handle it because of the class visibility? Would it make sense to produce info diagnostics for this one? It doesn't have to be part of this PR.

Reply:
Yup, should have mentioned this -- we produce a diagnostic for this. Hence, we need these suppressions:

https://github.com/dotnet/runtime/pull/91657/files#diff-81e6cde105e57e3f3b3a0107c173264b96aee943a0faa6b8f1bd72d7a57a43a3R333-R343